spark/v3.5/spark/src/main/java/org/apache/iceberg/spark/SparkTableUtil.java

```diff
@@ -711,7 +719,7 @@ public static void importSparkPartitions(
         spec,
         stagingDir,
         checkDuplicateFiles,
-        TableMigrationUtil.migrationService(parallelism));
+        executorService(parallelism));
```

Comment:
We already add `executeWith(ExecutorService service)` to `SnapshotTable` and `MigrateTable` interfaces and it's more testable than passing `parallelism`. Hence, I prefer to fix forward if not at high cost.

Reply:
How difficult is it for us to change this to a Supplier? I'm basically thinking that the current fix is basically creating a LazyExecutorService but we know actually passing a non-lazy executor service will always be broken. If this is the case shouldn't we pass through a Serializable Supplier for an Executor Service?

Based on your comments below it seems like this api is just broken as is. I think i may agree more with @nastra now that maybe we just remove the ability to pass through executor service? I'm not sure now

Reply:
@RussellSpitzer I'm going down the `SerializableSupplier` road. Please check whether it's ok before I fix comments, Spark 3.4 and Spark 3.3.

One side effect is I need to relocate `SerializableSupplier` from `iceberg-core` to `iceberg-api`.

Reply:
The reason why we didn't pass around parallelism in the first place is testing. There's no easy way to test the `ExecutorService` created and used in `TableMigrationUtil`.

Reply:
I thought the reason we have Executor Service is because we considered having multiple actions re-use the same service. So a implementer could control parallelism more easily.

Reply:
Ah yes this is in the API as well, https://github.com/apache/iceberg/blob/87a988fdda5fc68fd3b04947e000ef190fa54f61/api/src/main/java/org/apache/iceberg/actions/MigrateTable.java#L70